models/utils.go

```diff
@@ -0,0 +1,58 @@
+package models
+
+import (
+	"encoding/json"
+	"reflect"
+)
+
+func OmitEmpty(raw json.RawMessage) (json.RawMessage, error) {
+	var data map[string]any
+	if err := json.Unmarshal(raw, &data); err != nil {
+		return nil, err
+	} else {
+		StripEmptyEntries(data)
+		return json.Marshal(data)
+	}
+}
+
+func StripEmptyEntries(data map[string]any) {
+	for key, value := range data {
+		if isEmpty(reflect.ValueOf(value)) {
+			delete(data, key)
+		} else if nested, ok := value.(map[string]any); ok { // recursively strip nested maps
+			StripEmptyEntries(nested)
+		} else if slice, ok := value.([]any); ok {
+			data[key] = []any{}
+			for _, item := range slice {
+				if mapValue, ok := item.(map[string]any); ok {
+					StripEmptyEntries(mapValue)
+				} else {
+					data[key] = append(data[key].([]any), item)
+				}
+			}
+		} else {
+			data[key] = value
+		}
+	}
+}
```

Comment:
This logic sniffs fine to me from a first pass, but I'm noticing we didn't write a test for it and it's complex enough that having some automated tests would be a really good idea. Ideally a fairly complex structure that at minimum gives us 100% coverage of this function's lines.